Update SQL Scripts for Postgresql and MySQL and Code Cleanup

khu5h1 opened the pull request:
### Summary of Changes

- [x] Optimized Update SQL Scripts for Postgresql and MySQL.
- [x] Converted 2 Update SQL Scripts for Postgresql and MySQL into one.
- [x] Removed Label from Hidden Field.
- [x] Code Cleanup
- [x] Update String Value for Relative URL.
- [x] Increase the file version to 4.3 for Update SQL Scripts for Postgresql and MySQL.
- [x] Follow Coding Standards.
- [x] Resolved the PHPCS Errors from logs for getting the build tests passed.
- [x] Resolved the PHPCS Warnings from logs for getting the build tests passed.
- [x] Resolved the comments for code cleanup.
- [x] Used Standard Headings for the Views and removed the tours id from the title in the steps view.
- [x] Resolved the code review of PR #1.
- [x] Removed Overlay Option permanently and made it permanently true for every tour.

khu5h1 committed:
Update SQL Scripts for Postgresql and MySQL

khu5h1 committed:
Deleted one update sql script for Postgresql

khu5h1 committed:
Removed Label from Hidden Field

khu5h1 committed:
Code cleanup

bembelimen commented on administrator/components/com_admin/sql/updates/mysql/4.2.0-2022-06-16.sql:
```diff
@@ -21,6 +21,13 @@ CREATE TABLE IF NOT EXISTS `#__guidedtours` (
 ) ENGINE=InnoDB DEFAULT CHARSET=utf8mb4 DEFAULT COLLATE=utf8mb4_unicode_ci;
 
+--
+-- Dumping data for table `#__guidedtours`
```
Could you please increase the file version to 4.3?
Also check the other SQL what comment structure should be used.

bembelimen commented on administrator/components/com_admin/sql/updates/mysql/4.2.0-2022-06-16.sql:
```diff
@@ -43,4 +50,34 @@ CREATE TABLE IF NOT EXISTS `#__guidedtour_steps` (
+
+-- Add `plg_system_tour` to `#__extensions`
+INSERT INTO `#__extensions` (`package_id`, `name`, `type`, `element`, `folder`, `client_id`, `enabled`, `access`, `protected`, `locked`, `manifest_cache`, `params`, `custom_data`, `ordering`, `state`) VALUES
+(0, 'plg_system_tour', 'plugin', 'tour', 'system', 0, 1, 1, 0, 0, '', '{}', '', 15, 0);
```
Always empty line at the end of a file

bembelimen reviewed (changes requested):
When cleaned up, we can merge asap

khu5h1 committed:
Update String Value for Relative URL

khu5h1 committed:
Added extra lines in the end

khu5h1 committed:
Increase the file version to 4.3 for Update SQL Scripts for Postgresql and MySQL

khu5h1 committed:
Deleted 4.2  update sql script for Postgresql and Mysql

khu5h1 committed:
Code Cleanup

brianteeman commented on administrator/components/com_admin/sql/updates/postgresql/4.3.0-2022-07-30.sql:
```diff
@@ -0,0 +1,95 @@
+--
+
+INSERT INTO "#__guidedtours" ("id", "asset_id", "title", "description", "ordering", "extensions", "url", "overlay", "created", "created_by", "modified", "modified_by", "checked_out_time", "checked_out", "published") VALUES
+(1, 91, 'How to create a Guided Tour in Joomla Backend?', '<p>This Tour will show you that how you can create Guided Tour in Joomla Backend!</p>', 0, '[\"com_guidedtours\"]', 'administrator/index.php?option=com_guidedtours&view=tours', 1, '', 0, '', 0, '', 0, 1);
```
```suggestion
(1, 91, 'How to create a Guided Tour in Joomla Backend?', '<p>This Tour will show you how you can create a Guided Tour in the Joomla Backend!</p>', 0, '[\"com_guidedtours\"]', 'administrator/index.php?option=com_guidedtours&view=tours', 1, '', 0, '', 0, '', 0, 1);
```

brianteeman commented on plugins/system/tour/tour.php:
```diff
@@ -7,10 +7,11 @@
  * @category  Component
  * @package   Joomla.Administrator
  * @author    Joomla! <admin@joomla.org>
- * @copyright (C) 2013 Open Source Matters, Inc. <https://www.joomla.org>
+ * @copyright (C) 2022 Open Source Matters, Inc. <https://www.joomla.org>
  * @license   GNU General Public License version 2 or later; see LICENSE.txt
```
This is a plugin not a component. Please look at other plugins for the correct examples to follow

brianteeman commented on plugins/system/tour/tour.php:
```diff
@@ -32,143 +33,136 @@
+     * Load the language file on instantiation
+     *
+     * @var    boolean
+     * @since  3.1
```
```suggestion
     * @since __DEPLOY_VERSION__
 ```

Please be careful to update things when you have used copy/paste

brianteeman commented on administrator/language/en-GB/com_guidedtours.ini:
```diff
@@ -52,7 +52,7 @@ COM_GUIDEDTOURS_TITLE="Guided Tours"
-COM_GUIDEDTOURS_URL_DESC="Enter the Relative URL of the page from where you want to Start the Tour"
+COM_GUIDEDTOURS_URL_DESC="Enter the Relative URL of the page from where you want to Start the Tour e.g administrator/index.php?option=com_guidedtours&view=tours for guided tour relative URL."
 COM_GUIDEDTOURS_URL_LABEL="URL"
 COM_GUIDEDTOURS_XML_DESCRIPTION="Component for managing Guided Tours functionality in Joomla 4"
```
No need to say Joomla 4

khu5h1 committed:
Merge branch 'infra-guided-tour' of https://github.com/joomla-projects/gsoc22_guided-tour into cleanup

khu5h1 committed:
Code Cleanup

bembelimen committed:
Update drone signature for tests

khu5h1 committed:
Merge branch 'dev' of https://github.com/joomla-projects/gsoc22_guided-tour into cleanup

khu5h1 committed:
Cleanup

khu5h1 committed:
Resolved Warnings and followed coding standards for tests

khu5h1 committed:
Resolved Warning for Test

khu5h1 committed:
Added comment for PHPCS error

khu5h1 committed:
Used Standard Headings for the Views and removed the tours id from the title in the steps view.

brianteeman commented on administrator/components/com_admin/sql/updates/postgresql/4.3.0-2022-07-30.sql:
```diff
@@ -0,0 +1,96 @@
+--
+
+INSERT INTO "#__guidedtours" ("id", "asset_id", "title", "description", "ordering", "extensions", "url", "overlay", "created", "created_by", "modified", "modified_by", "checked_out_time", "checked_out", "published") VALUES
+(1, 91, 'How to create a Guided Tour in Joomla Backend?', '<p>This Tour will show you that how you can create Guided Tour in Joomla Backend!</p>', 0, '[\"com_guidedtours\"]', 'administrator/index.php?option=com_guidedtours&view=tours', 1, '', 0, '', 0, '', 0, 1);
```
this is not the correction I proposed

khu5h1 replied:
I have done the required changes. I have made changes in the script files and `extensions.sql` for Mysql and Postgresql.

brianteeman replied:
thank you

khu5h1 committed:
String value changes for tour and steps

khu5h1 committed:
Code cleanup

khu5h1 committed:
Used shorthand

khu5h1 committed:
Fixed Indentation of (C)

khu5h1 committed:
Fixed Indentation of step.xml

khu5h1 committed:
Code cleanup

khu5h1 committed:
Used Language Strings

khu5h1 committed:
Cleanup in step.xml and tour.xml

khu5h1 committed:
Removed Overlay Option and made it permanently true for every tour

khu5h1 committed:
Cleanup

khu5h1 committed:
PHPCS Fixes

khu5h1 committed:
Cleanup for tour.xml

khu5h1 committed:
Removed unused index for Postgresql

khu5h1 committed:
Added link of tour on steps view URL

khu5h1 committed:
Fixed the filtering of steps according to particular tour

richard67 commented:
SQL scripts (update scripts as well as extensions.sql) for PostgreSQL are still wrong. The initial author just copied the MySQL stuff and changed names quotes. But there are more differences than names quotes only between these 2 database types. E.g. in PostgreSQL there is no data type "datetime". "timestamp without time zone" has to be used instead.

I do not have the time now for a detailed review.

But it should be easy just to look at other existing SQL scripts and see what the differences to MySQL should be.

richard67 commented:
P.S.: The errors in the extensions.sql for PostgreSQL are the reason why api tests fail in drone (and system tests would fail, too).

richard67 committed:
Fix assets table changes in base.sql

- Fix wrong lft, rgt and level value
- Remove unused asset for plugin
- Adapt rgt value of the root item to the new asset

richard67 committed:
Adapt next asset id sequence number for PostgreSQL

When inserting values into a table with specifying the values for an auto-incremented ID column, the next value of the corresponding sequence has to be set on PostgreSQL databases.

richard67 committed:
Add missing values to insert for PostgreSQL

The values for the "ordering" and  "state" are missing here, which causes an SQL error when installing Joomla on a PostgreSQL database.

richard67 committed:
Don't use not existing asset id 91 on updates

On updates we do not create new assets, so we use asset_id zero for the records we insert on updates, see e.g. 4.0.0-2018-05-15.sql

richard67 committed:
Use INSERT IGNORE / ON CONFLICT DO NOTHING on updates

See https://github.com/joomla/joomla-cms/pull/37156 for details.

richard67 committed:
Update sequence numbers on PostgreSQL

Explanation see description of commit https://github.com/joomla-projects/gsoc22_guided-tour/commit/c1476f6414645af650d3a933833e4f284afd1b6a .

richard67 committed:
Consistency and code style fixes

- Some code style fixes
- Be consistent with how you specify primary keys in MySQL
- On update, insert new extensions with one insert statement and use ordering 0

richard67 committed:
Add missing extension record

Add missing extensions record for the plugin for new installations on PostgreSQL

richard67 committed:
Fix checked out columns to allow null values

- The checked_out and checked_out_time columns meanwhile allow null values and so also should not have a default value.
- Use  data types consistent with other tables for these columns.

richard67 committed:
Correct data types in PostgreSQL

- Use correct data types in PostgreSQL.
- Remove "state" column which is not  used for MySQL either.

richard67 committed:
Add CAN FAIL installer hint for PostgreSQL

See joomla/joomla-cms#37156 for details.

richard67 committed:
Fix for PostgreSQL

richard67 committed:
Fix extensions.sql for PostgreSQL

The soundex function has to be the last thing in this file.

richard67 committed:
Use correct values for created, modified and checked out columns

- Do not use wrong values for created and modified times.
- Do not use hardcoded user ID.

richard67 committed:
Fix my error from previous commit

richard67 committed:
Another invalid data type for PostgreSQL

richard67 committed:
Fix names quoting

- Hardcoded MySQL names quotes in this query lead to an SQL error on PostgreSQL databases
- Added other missing names quotes

richard67 committed:
Use double quotes inside SQL strings

richard67 committed:
Fix checked out column for PostgreSQL, too

richard67 committed:
Fix checked out columns in MySQL update SQL script

richard67 committed:
Add null value support to table classes

richard67 committed:
Remoce copy and paste remainders

Remove wrong code possibly resulting from copy and paset

richard67 committed:
SQL code style: Reserved words uppercase

richard67 committed:
Add PHPCS exception for underscore

richard67 committed:
Checked out for steps table

richard67 committed:
Add new core extensions to extensions helper

richard67 committed:
Fix for the database checker

The database checker for PostgreSQL seems to have a problem when the opening bracket is in a new line. This has to be fixed elsewhere, For now adapt to this behavior.

richard67 committed:
Adapt code style for some tables to the others

richard67 commented:
For fixes on SQL, especially for PostgreSQL, see #10 .

khu5h1 committed:
Added link of tour on steps view URL

khu5h1 committed:
Merge branch 'cleanup' of https://github.com/joomla-projects/gsoc22_guided-tour into cleanup-fix-sql-2022-08-13

khu5h1 committed:
Sync up changes with clean up branch

khu5h1 committed:
Merge pull request #10 from richard67/cleanup-fix-sql-2022-08-13

SQL fixes for PR #6
